src/radical/pilot/agent/scheduler/continuous_fifo.py

```diff
@@ -0,0 +1,51 @@
+
+__copyright__ = "Copyright 2013-2016, http://radical.rutgers.edu"
+__license__   = "MIT"
+
+
+from .continuous import Continuous
+
+# ------------------------------------------------------------------------------
+#
+# This is a simle extension of the Continuous scheduler which makes RP behave
+# like a FiFo: it will only really attempt to schedule units if all older units
+# (units with lower unit ID) have been scheduled.  As such, it relies on unit
+# IDs to be of the format `unit.%d'`.
+#
+class ContinuousFifo(Continuous):
+
+    # --------------------------------------------------------------------------
+    #
+    def __init__(self, cfg, session):
+
+        self.slots  = None
+        self._index = 0
+
+        Continuous.__init__(self, cfg, session)
+
+
+    # --------------------------------------------------------------------------
+    #
+    def _try_allocation(self, cu):
+
+        # check if the unit is up for scheduling
+        uid = cu['uid']
+        idx = int(uid.split('.')[1])
+
+        if idx > self._index + 1:
+            # nope - we let this attempt fail
+            self._log.debug('defer unit %s [%s]', uid, self._index)
+            return False
+
+        # yep: lets try for real
+        ret = Continuous.try_allocatio(self, cu)
+
+        if ret: 
+            # it worked!  Keep the new index
+            self.last = idx
```

Comment:
I think line 45 needs to be ```self._index =  idx```.

Although, in terms of naming, I am in favor of 'last' instead of 'index`.

Reply:
ack on both.  To be clear: I did not run this code, this is just for discussion right now, to check if this is a viable approach and actually solves the problem (short-term).

Reply:
:+1: